distributed/asyncio.py

```diff
@@ -0,0 +1,61 @@
+import asyncio
+from functools import wraps
+
+from tornado.gen import is_coroutine_function
+from tornado.platform.asyncio import BaseAsyncIOLoop, to_asyncio_future, to_tornado_future
+
+
+from .client import Client, Future
+
+
+def to_asyncio(method):
+    @wraps(method)
+    def convert(*args, **kwargs):
+        return to_asyncio_future(method(*args, **kwargs))
+    return convert
+
+
+class Asyncify(type):
+
+    def __new__(meta, cls, bases, attrs):
+        for base in bases:
+            for name, method in base.__dict__.items():
+                if name.startswith('_') and is_coroutine_function(method):
+                    attrs[name[1:]] = to_asyncio(method)
+        return type.__new__(meta, cls, bases, attrs)
+
+
+class AsyncFuture(Future, metaclass=Asyncify):
+
+    def __await__(self):
+        return self.result().__await__()
+
+
+class AsyncClient(Client, metaclass=Asyncify):
```

Comment:
Both clients are asynchronous, you might want to use a name prefix like AsyncIO instead.